src/common/util/jsonPatcher.js

```diff
@@ -182,8 +199,167 @@ define(['common/util/canon'], function (CANON) {
         return result;
     }
 
+    function _endsWith(str, pattern) {
+        var d = str.length - pattern.length;
+        return d >= 0 && str.lastIndexOf(pattern) === d;
+    }
+
+    function _isOvr(path) {
+        return path.indexOf('/ovr/') === 0;
+    }
+
+    function _isRelid(path) {
+        return RANDOM.isValidRelid(path.substring(1));
+    }
+
+    function _isGmePath(path) {
+        var relIds = path.split('/'),
+            result = false,
+            i;
+
+        for (i = 1; i < relIds.length; i += 1) {
+            if (RANDOM.isValidRelid(relIds[i]) === false) {
+                return false;
+            } else {
+                result = true;
+            }
+        }
+
+        return result;
+    }
+
+    function _inLoadOrUnload(res, gmePath) {
+        var pathPieces = gmePath.split('/'),
+            parentPath;
+
+        parentPath = gmePath;
+
+        do {
+            if (res.load[parentPath] || res.unload[parentPath]) {
+                return true;
+            }
+
+            pathPieces.pop();
+            parentPath = pathPieces.join('/');
+        } while (pathPieces.length > 1);
+
+        return false;
+    }
+
+    function _removeFromUpdates(res, gmePath) {
+        var updatesPath,
+            i;
+
+        updatesPath = Object.keys(res.update);
+        for (i = 0; i < updatesPath; i += 1) {
+            if (updatesPath[i].indexOf(gmePath) === 0) {
```

Comment:
A one line of comment would be great here. the indexOf operates on a string and not an array... this is like the startsWith function.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith